### PR TITLE
Add devauth as a dev dependency

### PR DESCRIPTION
**Pull request by DevScyu**

### CODE REVIEW

**Review by DevScyu — COMMENTED** [review]
what did the bot do o.o

File: build.gradle [hunk 1]
```diff
@@ -11,25 +11,19 @@ architectury {
 subprojects {
     apply plugin: "dev.architectury.loom"
 
+    repositories {
+        maven { url = "https://pkgs.dev.azure.com/djtheredstoner/DevAuth/_packaging/public/maven/v1" }
```
Comment (kristofbolyai) [inline build.gradle:15]: Can you just make this not a single line? (to match how the file looks like)

**Comment by kristofbolyai** [comment]
> what did the bot do o.o

By the way, the bot has a check to only run in non forked branches, so it is pretty much useless in it's current state as everyone uses a fork for PRs.

**Comment by DevScyu** [comment]
> > what did the bot do o.o
> 
> By the way, the bot has a check to only run in non forked branches, so it is pretty much useless in it's current state as everyone uses a fork for PRs.

Due to the git attributes file not declaring png files as binary, it tried to modify them and broke the images, hence the force pushes to fix it haha

File: build.gradle (hunk 1 above)
Comment (DevScyu) [reply build.gradle:15]: Followed formatting for repositories, also see: https://github.com/Wynntils/Artemis/blob/e96523ce91cd04c662bc6c6f21d7d2f04db002c1/build.gradle#L57

**Review by kristofbolyai — DISMISSED** [review]
Fine by me

File: .devauth/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+microsoft_accounts.json
```
Comment (magicus) [inline .devauth/.gitignore]: Missing newline

Comment (magicus) [reply .devauth/.gitignore]: Actually, is this correct or did you intend to modify the real `.gitignore` file?

File: .devauth/config.toml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+defaultEnabled = true
+defaultAccount = "main"
+
+[accounts.main]
+type = "microsoft"
```
Comment (magicus) [inline .devauth/config.toml]: Missing newline.

But does this work? I though DevAuth looked for ~/.devauth, not a .devauth folder in the mod root.

Edit: Ah, now I see where you set the devauth.configDir property.

File: .devauth/.gitignore (hunk 2 above)
Comment (magicus) [reply .devauth/.gitignore]: TIL: Apparently you can have multiple .gitignore files. I strongly recommend against this, though. Please add a `.devauth/microsoft_accounnds.json` rule to the top-level .gitignore file instead.

**Review by magicus — CHANGES_REQUESTED** [review]
Thanks for fixing this; it's been on my todo list for a while. Just some fine tuning left.

**Review by kristofbolyai — DISMISSED** [review]

File: .devauth/config.toml (hunk 3 above)
Comment (kristofbolyai) [reply .devauth/config.toml]: Did you forget to add the newline?

**Review by kristofbolyai — DISMISSED** [review]

File: gradle.properties [hunk 4]
```diff
@@ -7,6 +7,8 @@ parchment_version=2022.05.02
 fabric_loader_version=0.13.3
 fabric_version=0.51.1+1.18.2
 forge_version=1.18.2-40.1.0
+# DevAuth version
+devauth_version=1.0.0
 # Note: Must use same version of eventbus as is used in the above forge version!
 forge_eventbus_version=5.0.+
```
Comment (kristofbolyai) [inline gradle.properties]: Actually just noticed this, can we not separate the forge event bus and forge version? They are closely related and a change to one may cause the other one to be outdated as well.

Comment (magicus) [reply gradle.properties]: Agree. Other than that, it looks good.

Comment (DevScyu) [reply gradle.properties]: whoops

**Review by kristofbolyai — APPROVED** [review]

**Review by magicus — APPROVED** [review]

**Comment by kristofbolyai** [comment]
@DevScyu This does not seem to work properly, had to set `"-Ddevauth.enabled=true"` to make it enabled, so it seems it ignored the config?

**Pull request merged**